### ISSUE TITLE
Date picker: Implement conditional formatting:

### ISSUE THREAD

**Issue by jaclinec** [uswds/uswds#5629]
1. Automatically add `0` for single-digit dates.
2. Automatically enter slashes so the user does not have to type them.

**Comment by jaclinec** [comment]
Closing per @thisisdano. This issue is captured in #5534 and work will be tracked there.
